client/core/core.go

```diff
@@ -128,6 +128,24 @@ func (dc *dexConnection) hasOrders(assetID uint32) bool {
 	return false
 }
 
+// hasActiveOrders checks whether there are any active orders for the dexConnection.
+func (dc *dexConnection) hasActiveOrders() bool {
+	dc.tradeMtx.RLock()
+	defer dc.tradeMtx.RUnlock()
+
+	for _, trade := range dc.trades {
+		for _, match := range trade.matches {
+			if match.MetaData.Status < order.MakerRedeemed {
+				return true
+			}
+		}
```

Comment:
You'll need to lock the `(*trackedTrade).matchMtx` while iterating `trade.matches`. I'd suggest you move this loop to a method on the `trackedTrade` and use a `defer` to unlock.